framework/helpers/BaseJson.php

```diff
@@ -179,16 +192,28 @@ protected static function processData($data, &$expressions, $expPrefix)
                 $data = $data->toArray();
             } elseif ($data instanceof \SimpleXMLElement) {
                 $data = (array) $data;
+
+                // Avoid empty elements to be returned as array.
+                // Not breaking BC because empty array was always cast to stdClass before.
+                $revertToObject = true;
             } else {
+                /*
+                 * $data type is changed to array here and its elements will be processed further
+                 * We must cast $data back to object later to keep intended dictionary type in JSON.
+                 * Revert is only done when keepObjectType flag is provided to avoid breaking BC
+                 */
+                $revertToObject = self::$keepObjectType;
```

Comment:
Call property as `static` instead `self` in `Base*` classes - users can override/extend class `Json` in extension

Reply:
Changed it, thanks